dask/array/rechunk.py

```diff
@@ -205,7 +203,7 @@ def rechunk(x, chunks):
 
     crossed = intersect_chunks(x.chunks, chunks)
     x2 = dict()
-    temp_name = next(rechunk_names)
+    temp_name = 'rechunk-' + tokenize((x.name, chunks))
```

Comment:
Depends on what you consider meaningful. The dispatch function is faster for all cases than an if statement with more than 3 cases (on my computer), so the real overhead then is how long it takes to normalize each argument. It'd be a couple `us` for each operation, which I guess isn't something we probably care about.